File: python/armstrong.py
# ======================================================
# 📘 Practical – 2 : Armstrong Number
# ======================================================

# 🔹 Aim / Objective:
#   To write a Python program to check whether a number 
#   is an Armstrong Number or not.

# 🔹 Tools Required:
#   - Python 3.x Interpreter
#   - Text Editor / IDE (IDLE, VS Code, PyCharm etc.)

# 🔹 Algorithm:
#   1. Start program
#   2. Input a number from user
#   3. Count total digits of the number
#   4. Extract each digit one by one
#   5. Raise each digit to the power of total digits
#   6. Add all powered values together
#   7. Compare with the original number
#   8. If same → Armstrong, else → Not Armstrong
#   9. Stop program

# 🔹 Steps:
#   Step 1: Start the program
#   Step 2: Take input from the user
#   Step 3: Find total digits (order)
#   Step 4: Initialize sum = 0
#   Step 5: Copy original number into temp variable
#   Step 6: Extract each digit using modulus (%)
#   Step 7: Raise digit to 'order' power and add to sum
#   Step 8: Repeat until all digits processed
#   Step 9: Compare sum with original number
#   Step 10: Print result
#   Step 11: Stop program

# ======================================================
# 🔹 Code
# ======================================================

# Step 1: Take input from user
num = int(input("Enter a number: "))

# Step 2: Count total digits
order = len(str(num))

# Step 3: Initialize sum = 0
sum_val = 0

# Step 4: Copy number into temp variable
temp = num

# Step 5: Loop through digits
while temp > 0:
    digit = temp % 10            # extract last digit
    sum_val += digit ** order    # add digit^order to sum
    temp //= 10                  # remove last digit

# Step 6: Compare sum with original number
if num == sum_val:
    print(num, "is an Armstrong Number")
else:
    print(num, "is NOT an Armstrong Number")

# ======================================================
# 🔹 Sample Output
# ======================================================
# Enter a number: 153
# 153 is an Armstrong Number
#
# Enter a number: 123
# 123 is NOT an Armstrong Number
# ======================================================

# 🔹 Result:
# The program successfully checks and displays whether 
# the entered number is an Armstrong Number or not.
# ======================================================

File: PHP/armstrong.php
<?php
/*
=====================================================
PRACTICAL: PHP Program to Check Armstrong Number
=====================================================

AIM:
- To write a PHP program that checks whether
  a given number is an Armstrong number or not.

OBJECTIVE:
- To read a number from user.
- To calculate sum of cubes of its digits.
- If sum equals original number → Armstrong.
- Else → Not Armstrong.

TOOLS:
- PHP 7+/8+
- Any Web Server (XAMPP/WAMP/LAMP)

ALGORITHM:
1) Accept a number from user.
2) Store original number in temp variable.
3) Initialize sum = 0.
4) While number > 0:
   - digit = number % 10
   - sum = sum + (digit^3)
   - number = number / 10
5) If sum == original → Armstrong.
6) Else → Not Armstrong.

STEPS TO RUN:
- Save file as: armstrong.php
- Open in browser: http://localhost/armstrong.php
- Enter number and check.

=====================================================
PHP CODE (armstrong.php)
=====================================================
*/
?>

<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <title>Armstrong Number Check</title>
</head>
<body>
  <h1>Armstrong Number Check (PHP)</h1>

  <form method="post">
    <label>Enter a number:</label><br>
    <input type="number" name="num" min="0" required>
    <button type="submit">Check</button>
  </form>

<?php
if ($_SERVER["REQUEST_METHOD"] === "POST") {
    $num = (int)$_POST["num"];
    $original = $num;
    $sum = 0;

    while ($num > 0) {
        $digit = $num % 10;
        $sum += $digit * $digit * $digit; // cube of digit
        $num = (int)($num / 10);
    }

    if ($sum == $original) {
        echo "<p>$original is an Armstrong Number.</p>";
    } else {
        echo "<p>$original is NOT an Armstrong Number.</p>";
    }
}
?>

<?php
/*
================= SAMPLE OUTPUT =================

Input: 153  
Output: 153 is an Armstrong Number.  
(1³ + 5³ + 3³ = 1 + 125 + 27 = 153)

Input: 123  
Output: 123 is NOT an Armstrong Number.

=================================================
RESULT:
- Successfully implemented a PHP program that
  checks Armstrong number.
=================================================
*/
?>
</body>
</html>
